packages/expression-evaluator/lib/expressions/SpecialOperator.ts

```diff
@@ -1,14 +0,0 @@
-import type { FunctionApplication } from '../functions/OverloadTree';
-import type { Expression, SpecialOperatorExpression } from './Expressions';
-import {
-  ExpressionType,
-} from './Expressions';
-
-export class SpecialOperator implements SpecialOperatorExpression {
```

Comment:
@rubensworks I was able to delete this expression type as a result of our previous discussion.

I think a similar approach can be taken for the namedFunctions: https://github.com/jitsedesmet/comunica/blob/chore/bussify-all-function/packages/expression-evaluator/lib/expressions/Named.ts

The difference is that namedFunctions also have a `name` presented as a namedNode.
SparqlOperator obviously do not have this, but they could contain a name as a string (like `=`). I would like to also generalize named functions to a string (so that internally, they are approached just like a function with a URI as name). I think this will make for cleaner code. I also don't see an issue with this generalization because we want to look beyond SPARQL anyway.